### PR TITLE
MM-55733 add request context to Channelstore.Save method

### PR DESCRIPTION
**Pull request by titanventura**
#### Summary

Added request context to channelstore.save method

#### Ticket Link

Github Issue: #26023 
JIRA: https://mattermost.atlassian.net/browse/MM-55733

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by isacikgoz — APPROVED** [review]
LGTM!

File: server/channels/app/teams/teams.go [hunk 1]
```diff
@@ -56,7 +56,8 @@ func (ts *TeamService) createDefaultChannels(teamID string) ([]*model.Channel, e
 		// and let the subscribers do the job, in this case it would be the channels service.
 		// Currently we are adding services to the server and because of that we are using
 		// the channel store here. This should be replaced in the future.
-		if _, err := ts.channelStore.Save(channel, *ts.config().TeamSettings.MaxChannelsPerTeam); err != nil {
+		rctx := request.EmptyContext(ts.log)
+		if _, err := ts.channelStore.Save(rctx, channel, *ts.config().TeamSettings.MaxChannelsPerTeam); err != nil {
```
Comment (hanzei) [inline server/channels/app/teams/teams.go:60]: Instead of creating a new context, can we change `CreateTeam` and `createDefaultChannels` to accept one?

**Review by hanzei — CHANGES_REQUESTED** [review]
Nice work, thanks @titanventura :+1: 

I have one request regarding the team service, see below.

**Comment by hanzei** [comment]
The PR requires changes in the closed-source Enterprise repository. I will submit them once this PR is approved.

File: server/channels/app/teams/teams.go (hunk 1 above)
Comment (titanventura) [reply server/channels/app/teams/teams.go]: sure @hanzei good catch ! missed it.

File: server/channels/app/teams/service.go [hunk 2]
```diff
@@ -18,6 +19,7 @@ type TeamService struct {
 	wh           WebHub
 	config       func() *model.Config
 	license      func() *model.License
+	log          mlog.LoggerIFace
```
Comment (hanzei) [inline server/channels/app/teams/service.go:22]: Is this still needed?

File: server/channels/app/teams/service.go [hunk 3]
```diff
@@ -30,6 +32,7 @@ type ServiceConfig struct {
 	WebHub       WebHub
 	ConfigFn     func() *model.Config
 	LicenseFn    func() *model.License
+	Logger       mlog.LoggerIFace
```
Comment (hanzei) [inline server/channels/app/teams/service.go:35]: Same question here

**Review by hanzei — APPROVED** [review]
Thanks for updating the PR :+1:

**Review by hanzei — APPROVED** [review]
LGTM :+1:

**Comment by hanzei** [comment]
@titanventura Would you please merge `master` into your PR? Then we can merge the PR.

**Comment by hanzei** [comment]
Awesome, we are good to go :rocket: Thanks for the great work @titanventura

**Pull request merged**